src/client/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -37,16 +37,20 @@ internal AbstractApplicationBuilder(ApplicationConfiguration configuration)
         /// or setting the Agent.
         /// </summary>
         /// <param name="httpClientFactory">HTTP client factory</param>
+        /// <param name="retryOnceOn5xx">Configures MSAL to retry on 5xx server errors. When enabled (on by default), MSAL will wait 1 second after recieving
+        /// a 5xx error and then retry the http request again.</param>
         /// <remarks>MSAL does not guarantee that it will not modify the HttpClient, for example by adding new headers.
         /// Prior to the changes needed in order to make MSAL's httpClients thread safe (https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2046/files),
         /// the httpClient had the possibility of throwing an exception stating "Properties can only be modified before sending the first request".
         /// MSAL's httpClient will no longer throw this exception after 4.19.0 (https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/releases/tag/4.19.0)
         /// see (https://aka.ms/msal-httpclient-info) for more information.
+        /// If you only want to configure the retryOnceOn5xx parameter, set httpClientFactory to null and MSAL will use the default Http client.
         /// </remarks>
         /// <returns>The builder to chain the .With methods</returns>
-        public T WithHttpClientFactory(IMsalHttpClientFactory httpClientFactory)
+        public T WithHttpClientFactory(IMsalHttpClientFactory httpClientFactory, bool retryOnceOn5xx = true)
```

Comment:
I believe we said in the past that we'd add new overloads instead of adding new default params.

Reply:
Yes I think it makes sense. I was following what was in this comment so I wasn't sure.
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/3608#discussion_r953699033

I can update